### ISSUE TITLE
autograd.Function <-> functorch interaction followup items

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#90224]
Recording them here so I don't forget. For the stack over at https://github.com/pytorch/pytorch/pull/90077

- [x] Better error message when someone breaks the vmap staticmethod API contract
- [x] figure out once_differentiable handling 
- [x] autograd.Function.apply: link documentation from error message
- [x] custom_function_call_vmap: link docs when ready
- [x] cleanup: generate_vmap_rule should exist on autograd.Function
- [x] cleanup: delete the feature flag
- [x] https://github.com/pytorch/pytorch/issues/91451
- [x] Better name for the Generated autograd.Function in custom_function_call_grad
- [x] autograd.Function.apply: get rid of the circular import
- [x] Update comments about get_fwd_mode: https://github.com/pytorch/pytorch/pull/90240#issuecomment-1349596445
- [x] name bikeshedding https://github.com/pytorch/pytorch/pull/90418#discussion_r1047721208

@soulitzer if you think I missed anything

cc @ezyang @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @Chillee @samdow @soumith

**Comment by zou3519** [comment]
Talked with Alban offline on "figure out once_differentiable handling" -- that is resolved, the functorch transforms give the consistent behavior on once_differentiable as torch.autograd.grad. torch.autograd.grad considers the first-order gradients unrelated to the original inputs so it raises an error on second-order gradient computation; functorch returns zero-gradients in this case. If we want we can improve once_differentiable in the future in a way that the first-order gradients are related to the original inputs AND raise an error when we backprop through them.

**Comment by zou3519** [comment]
All items completed.
